### PR TITLE
Add check for the differences between time columns

### PR DESCRIPTION
**Pull request by mg-aquaq**
This can be used to measure latency between timestamps from the exchange and timestamps from the ticker plant.

### CODE REVIEW

**Comment by mg-aquaq** [comment]
There's no config for this, as I tested it with a different version of quote to the one in TorQ, where I added an "exchange time" column.

File: code/dqc/timediff.q [hunk 1]
```diff
@@ -0,0 +1,12 @@
+\d .dqc
+
+/ Takes a table name tn and two column names ca and cb, as well as a percentage
+/ pt and a tolerance tl in milliseconds.
+timediff:{[tn;ca;cb;pt;tl]
+  .lg.o[`timediff;"HELLO"];
```
Comment (ThomasSmyth) [inline code/dqc/timediff.q:6]: For debugging?

File: code/dqc/timediff.q [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  .lg.o[`timediff;"Checking the time differences in columns ",(", "sv string(ca;cb))," of table ",(string tn)];
+  $[tl>re:1-(sum a)%count a:?[tn;();();(>;(0D+tl*00:00:00.001);(-;ca;cb))];
+    (1b;"No major problem with data flow");
```
Comment (ThomasSmyth) [inline code/dqc/timediff.q:9]: Adding logging for these output messages too.

File: code/dqc/timediff.q (hunk 1 above)
Comment (mg-aquaq) [reply code/dqc/timediff.q:6]: oops, shouldn't have let that slip through

File: code/dqc/timediff.q (hunk 2 above)
Comment (mg-aquaq) [reply code/dqc/timediff.q:9]: As in.. save the result to a variable, log it, then return  the variable?

File: code/dqc/timediff.q [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    (0b;"ERROR: ",(string re*100),"% of differences between columns ",(string ca),", ",m:(string cb)," are more than ",(string tl)," milliseconds.")
+    ]
```
Comment (ThomasSmyth) [inline code/dqc/timediff.q:11]: Something like this should do
```suggestion
  msg:$[c:tl>re:1-(sum a)%count a:?[tn;();();(>;(0D+tl*00:00:00.001);(-;ca;cb))];
    "No major problem with data flow";
    "ERROR: ",(string re*100),"% of differences between columns ",(string ca),", ",m:(string cb)," are more than ",(string tl)," milliseconds."
    ];
    .lg.o[`timediff;msg];
    (c;msg)
```

File: code/dqc/timediff.q [hunk 4]
```diff
@@ -0,0 +1,13 @@
+\d .dqc
+
+/ Takes a table name tn and two column names ca and cb, as well as a percentage
+/ pt and a tolerance tl in milliseconds.
+timediff:{[tn;ca;cb;pt;tl]
+  .lg.o[`timediff;"Checking the time differences in columns ",(", "sv string(ca;cb))," of table ",(string tn)];
+  ot:$[tl>re:1-(sum a)%count a:?[tn;();();(>;(0D+tl*00:00:00.001);(-;ca;cb))];
```
Comment (jonnypress) [inline code/dqc/timediff.q:7]: ?[tn;();();(>;(0D+tl*00:00:00.001);(-;ca;cb))]
What is this... an exec? 
Does this work on an ondisk table? Does it need to?

File: code/dqc/timediff.q [hunk 5]
```diff
@@ -0,0 +1,13 @@
+\d .dqc
+
+/ Takes a table name tn and two column names ca and cb, as well as a percentage
+/ pt and a tolerance tl as a timespan (eg 0D00:00:00.001).
+timediff:{[tn;ca;cb;pt;tl]
+  .lg.o[`timediff;"Checking the time differences in columns ",(", "sv string(ca;cb))," of table ",(string tn)];
+  ot:$[tl>re:1-(sum a)%count a:tl>(tn ca)-tn cb;
+    (1b;"No major problem with data flow");
+    (0b;"ERROR: ",(string re*100),"% of differences between columns ",(string ca),", ",m:(string cb)," are more than ",(string tl)," milliseconds.")
```
Comment (jonathonmcmurray) [inline code/dqc/timediff.q:9]: does the "miiliseconds" at the end of the message still make sense if `t1` is a timespan now? Might be OK to just omit that

Comment (mg-aquaq) [reply code/dqc/timediff.q:9]: You're right, it should always be a time span now, will change that.

File: code/dqc/timediff.q [hunk 6]
```diff
@@ -0,0 +1,13 @@
+\d .dqc
+
+/ Takes a table name tn and two column names ca and cb, as well as a percentage
+/ pt and a tolerance tl as a timespan (eg 0D00:00:00.001).
+timediff:{[tn;ca;cb;pt;tl]
+  .lg.o[`timediff;"Checking the time differences in columns ",(", "sv string(ca;cb))," of table ",(string tn)];
+  ot:$[tl>re:1-(sum a)%count a:tl>(tn ca)-tn cb;
```
Comment (jonathonmcmurray) [inline code/dqc/timediff.q:7]: is this comparison correct? if I'm reading it right, you're comparing the tolerance timespace to the percentage of times that differ by greater than the tolerance. Should this be

```suggestion
  ot:$[pt>re:1-(sum a)%count a:tl>(tn ca)-tn cb;
```

?

File: code/dqc/timediff.q [hunk 7]
```diff
@@ -0,0 +1,13 @@
+\d .dqc
+
+/ Takes a table name tn and two column names ca and cb, as well as a percentage
+/ pt and a tolerance tl as a timespan (eg 0D00:00:00.001).
+timediff:{[tn;ca;cb;pt;tl]
+  .lg.o[`timediff;"Checking the time differences in columns ",(", "sv string(ca;cb))," of table ",(string tn)];
+  ot:$[pt>re:1-(sum a)%count a:tl>(tn ca)-tn cb;
```
Comment (jonathonmcmurray) [inline code/dqc/timediff.q:7]: If you switch the more than to a less than, can you get rid of the `1-` like this?

```suggestion
  ot:$[pt>re:(sum a)%count a:tl<(tn ca)-tn cb;
```

Comment (mg-aquaq) [reply code/dqc/timediff.q:7]: Probably, but I've just merged it

**Pull request merged**